File: python/armstrong_numbers.py
# How can you make this more scalable and reusable later?
def find_armstrong_numbers(numbers):
    armstrong=[]
    sum = 0
    for num in range(0, numbers[-1]+1):
        num_str = str(num)
        l = len(num_str)
        for digit in num_str: 
            sum +=int(digit)**l
        if sum == num:
            armstrong.append(num)
        sum=0
    print(armstrong)
    return(armstrong)
find_armstrong_numbers([1000])

File: js/armstrongNumbers.js
// How can you make this more scalable and reusable later?
function findArmstrongNumbers(numbers) {
   let armstrong=[];
   let  sum = 0;
   for (let num = 0; num <=numbers[numbers.length-1]; num++) {
        let num_str = num.toString();
        let l = num_str.length;
        for (let i = 0; i < num_str.length; i++) {
            sum += Math.pow(parseInt(num_str[i]), l);
        }
        if (sum === num) {
            armstrong.push(num);
        }
        sum=0;
    }
    // console.log(armstrong);
    return armstrong;
}
findArmstrongNumbers([1000]);

module.exports = {findArmstrongNumbers}
